### PR TITLE
[extension-selenium] Add case ignoring natural and revers order

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

File: vividus-extension-selenium/src/test/java/org/vividus/bdd/steps/ui/model/StringSortingOrderTests.java [hunk 1]
```diff
@@ -27,18 +28,28 @@
...
     @ParameterizedTest
     @MethodSource("dataProvider")
-    void shouldReturnSortingComparator(StringSortingOrder sortingOrder, Comparator<? extends Comparable> comparator)
+    void shouldReturnSortingComparator(StringSortingOrder sortingOrder, List<String> expected)
     {
-        assertEquals(sortingOrder.getSortingType(), comparator);
+        List<String> sorterd = List.of(KEY_1, KEY_2, KEY_3, KEY_4).stream()
+                                                                  .sorted(sortingOrder.getSortingType())
+                                                                  .collect(Collectors.toList());
+        assertEquals(expected, sorterd);
```
Comment (ikalinin1) [inline vividus-extension-selenium/src/test/java/org/vividus/bdd/steps/ui/model/StringSortingOrderTests.java:53]: List.of(KEY_1, KEY_2, KEY_3, KEY_4).sort(sortingOrder.getSortingType()) ?

Comment (uarlouski) [reply vividus-extension-selenium/src/test/java/org/vividus/bdd/steps/ui/model/StringSortingOrderTests.java:53]: it will fail because List.of is immutable

Comment (ikalinin1) [reply vividus-extension-selenium/src/test/java/org/vividus/bdd/steps/ui/model/StringSortingOrderTests.java:53]: Arrays.asList(...) then? :)

Comment (ikalinin1) [reply vividus-extension-selenium/src/test/java/org/vividus/bdd/steps/ui/model/StringSortingOrderTests.java:53]: `sorterd` -> `sorted`

Comment (uarlouski) [reply vividus-extension-selenium/src/test/java/org/vividus/bdd/steps/ui/model/StringSortingOrderTests.java]: sort returns void, so imho sort via stream looks good

File: docs/modules/plugins/partials/generic-ui-steps.adoc [hunk 2]
```diff
@@ -360,7 +360,7 @@ Then elements located `$locator` are sorted by text in $sortingOrder order
 ==== *_Parameters_*
 
 . `$locator` - <<_locator>>
-. `$sortingOrder` - either https://www.computerhope.com/jargon/a/ascendin.htm[ASCENDING] or https://www.computerhope.com/jargon/d/descorde.htm[DESCENDING] order
+. `$sortingOrder` - either https://www.computerhope.com/jargon/a/ascendin.htm[ASCENDING] or https://www.computerhope.com/jargon/d/descorde.htm[DESCENDING] or CASE_INSENSITIVE_ASCENDING or CASE_INSENSITIVE_DESCENDING order
```
Comment (valfirst) [inline docs/modules/plugins/partials/generic-ui-steps.adoc:363]: please format the possible values in list-like view
also please provide possble user-friendly values like `case-insensitive ascending`

**Review by valfirst — CHANGES_REQUESTED** [review]

**Comment by valfirst** [comment]
`revers` -> `reverse`

File: vividus-extension-selenium/src/test/java/org/vividus/bdd/steps/ui/model/StringSortingOrderTests.java (hunk 1 above)
Comment (ikalinin1) [reply vividus-extension-selenium/src/test/java/org/vividus/bdd/steps/ui/model/StringSortingOrderTests.java]: JMH time :)

Comment (ikalinin1) [reply vividus-extension-selenium/src/test/java/org/vividus/bdd/steps/ui/model/StringSortingOrderTests.java]: https://stackoverflow.com/questions/49798129/what-is-more-efficient-sorted-stream-or-sorting-a-list
image

**Review by valfirst — APPROVED** [review]

File: vividus-extension-selenium/src/test/java/org/vividus/bdd/steps/ui/model/StringSortingOrderTests.java (hunk 1 above)
Comment (uarlouski) [reply vividus-extension-selenium/src/test/java/org/vividus/bdd/steps/ui/model/StringSortingOrderTests.java]: okay, but
- this is just a unit test
- lets stop using streams where we don't need parallel computations?

**Review by ikalinin1 — APPROVED** [review]

**Pull request merged**